src/models/community_business.ts

```diff
@@ -279,5 +285,104 @@ export const CommunityBusinesses: CommunityBusinessCollection & CustomMethods =
 
     return query;
   },
+
+  async getVisitActivities (client: Knex, o: CommunityBusiness, d?: Day) {
+    const baseQuery = client('visit_activity')
+      .innerJoin(
+        'visit_activity_category',
+        'visit_activity_category.visit_activity_category_id',
+        'visit_activity.visit_activity_category_id')
+      .select({
+        id: 'visit_activity_id',
+        name: 'visit_activity_name',
+        category: 'visit_activity_category.visit_activity_category_name',
+        monday: 'monday',
+        tuesday: 'tuesday',
+        wednesday: 'wednesday',
+        thursday: 'thursday',
+        friday: 'friday',
+        saturday: 'saturday',
+        sunday: 'sunday',
+        createdAt: 'created_at',
+        modified_at: 'modified_at',
+      })
+      .where({ organisation_id: o.id, deleted_at: null });
+
+    const query = d
+      ? baseQuery.where({ [d]: true })
+      : baseQuery;
+
+    return query;
+  },
+
+  async addVisitActivity (client: Knex, v: Partial<VisitActivity>, c: Partial<CommunityBusiness>) {
+    const [res] = await client('visit_activity')
+      .insert({
+        visit_activity_name: v.name,
+        visit_activity_category_id: client('visit_activity_category')
+          .select('visit_activity_category_id')
+          .where({ visit_activity_category_name: v.category }),
+        organisation_id: c.id,
+      })
+      .returning([
+        'visit_activity_id AS id',
+        'visit_activity_name AS name',
+        'monday',
+        'tuesday',
+        'wednesday',
+        'thursday',
+        'friday',
+        'saturday',
+        'sunday',
+        'created_at AS createdAt',
+        'modified_at as modifiedAt',
+      ]);
+    return res || null;
```

Comment:
Where did we land on this being inconsistent with the type declaration (and possibly papering over a failed insert)?